zipper/broadcast/broadcast_group.go

```diff
@@ -102,7 +102,7 @@ func fetchRequestToKey(prefix string, request *protov3.MultiFetchRequest) string
 	return string(key)
 }
 
-func (bg *BroadcastGroup) doSingleFetch(ctx context.Context, logger *zap.Logger, client types.ServerClient, request *protov3.MultiFetchRequest, doneCh chan<- string, resCh chan<- *types.ServerFetchResponse) {
+func (bg *BroadcastGroup) doSingleFetch(ctx context.Context, logger *zap.Logger, client types.ServerClient, requests []*protov3.MultiFetchRequest, doneCh chan<- string, resCh chan<- *types.ServerFetchResponse) {
```

Comment:
I do not really understand why it was required to move for-loop here. Also I think that in that case 'doSingleFetch' must be renamed to something else, as it's not a single MultiFetchRequest anymore. But that can be done after PR is merged.